### PR TITLE
[Wf-Diagnostics] introduce emitter interface in w/f diagnostics

### PR DESCRIPTION
**Pull request by sankari165**
**What changed?**
Workflow Diagnostics has a emitter interface that produces messages to a kafka topic

**Why?**
this emitter will be used to emit usage logs from workflow diagnostics runs to a pinot table 

**How did you test it?**
unit tests

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: service/worker/diagnostics/analytics/emitter.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+type emitter struct {
+	producer messaging.Producer
+}
+
+type NewEmitterParams struct {
+	Producer messaging.Producer
+}
+
+func NewEmitter(p NewEmitterParams) DataEmitter {
```
Comment (dkrotx) [inline service/worker/diagnostics/analytics/emitter.go:44]: do you see there will be a lot (>3) args to the `NewEmitter`. Is it worth introducing `NewEmitterParams` now?
in any way: nit - to rename `NewEmitterParams` to EmitterParams or something - it is not typical to call structure-holding-all-arguments as `NewSomething`.

File: service/worker/diagnostics/analytics/emitter.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	msg[Identity] = data.Identity
+	msg[SatisfactionFeedback] = data.SatisfactionFeedback
+	msg[IssueType] = data.IssueType
+	msg[DiagnosticsWfID] = data.DiagnosticsWorkflowID
+	msg[DiagnosticsWfRunID] = data.DiagnosticsRunID
+	msg[Environment] = data.Environment
+	msg[DiagnosticsStartTime] = data.DiagnosticsStartTime
+	msg[DiagnosticsEndTime] = data.DiagnosticsEndTime
+
+	serializedMsg, err := json.Marshal(msg)
```
Comment (dkrotx) [inline service/worker/diagnostics/analytics/emitter.go:64]: why not just serialize `WfDiagnosticsUsageData`? It that to have some compatibility?

Comment (sankari165) [reply service/worker/diagnostics/analytics/emitter.go:64]: am following whats done in visibility. will double check with Neil too

File: service/worker/diagnostics/analytics/emitter.go (hunk 1 above)
Comment (sankari165) [reply service/worker/diagnostics/analytics/emitter.go]: regarding more args, I am expecting logging and metrics.

File: service/worker/diagnostics/analytics/emitter.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	if err != nil {
+		return err
+	}
+
+	pinotMsg := &indexer.PinotMessage{
```
Comment (taylanisikdemir) [inline service/worker/diagnostics/analytics/emitter.go:69]: shouldn't this layer be visibility-store agnostic?

**Pull request merged**